test/unmarshal_test.go

```diff
@@ -0,0 +1,28 @@
+package test
+
+import (
+	"github.com/coinbase/kryptology/pkg/signatures/bls/bls_sig"
+	"github.com/stretchr/testify/require"
+	"testing"
+)
+
+func TestUnmarshaling(t *testing.T) {
+	blsScheme := bls_sig.NewSigEth2()
+	_, sks, err := blsScheme.ThresholdKeygen(3, 5)
+	require.NoError(t, err)
+
+	for _, sk := range sks {
+		b1, err := sk.MarshalBinary()
+		require.NoError(t, err)
+
+		// UnmarshalBinary giving empty sk.value
+		sk1 := new(bls_sig.SecretKeyShare)
+		err = sk1.UnmarshalBinary(b1)
+		require.NoError(t, err)
+
+		// minus 1 because of identifier
+		zeros := make([]byte, len(b1)-1)
+		b2, err := sk1.MarshalBinary()
+		require.Equal(t, zeros, b2[1:])
```

Comment:
I see, I wish you had written a test that would fail on incorrect behaviour ;)

Reply:
@dB2510 the fix in your open PR looks good to me. Please move this test to `[lib_test.go](https://github.com/coinbase/kryptology/blob/master/pkg/signatures/bls/bls_sig/lib_test.go)` and change it so that it fails if the unmarshaling does not do the right job and I'll approve.

Thanks for reporting and fixing the issue :)

Reply:
@Arash-Afshar Sure, will move this test to lib_test.go.